ptvsd/wrapper.py

```diff
@@ -1286,21 +1279,30 @@ def _stop_event_loop(self):
 
     def start(self, threadname):
         super(VSCodeMessageProcessor, self).start(threadname)
-        self._subprocess_notifier_thread = _util.new_hidden_thread('SubprocessNotifier', self._subprocess_notifier)
-        self._subprocess_notifier_thread.start()
+        if options.multiprocess:
+            self._subprocess_notifier_thread = _util.new_hidden_thread('SubprocessNotifier', self._subprocess_notifier)
+            self._subprocess_notifier_thread.start()
 
     def close(self):
         super(VSCodeMessageProcessor, self).close()
-        self._subprocess_notifier_thread.join()
+        if options.multiprocess:
+            self._subprocess_notifier_thread.join()
 
     def _subprocess_notifier(self):
         while not self.closed:
             try:
-                subprocess_info = multiproc.subprocess_queue.get(block=False, timeout=0.1)
+                subprocess_request, subprocess_response = multiproc.subprocess_queue.get(block=False, timeout=0.1)
             except queue.Empty:
+                continue
+
+            try:
+                self.send_event('ptvsd_subprocess', **subprocess_request)
```

Comment:
Note that this is still send_event, despite our discussion. The reason is that the old IPC JSON implementation does not support sending requests, only handling them. We should revisit this after refactoring.